crates/renderling/src/stage/gltf_support.rs

```diff
@@ -1002,58 +991,67 @@ impl GltfDocument {
-                    gltf::khr_lights_punctual::Kind::Spot {
-                        inner_cone_angle,
-                        outer_cone_angle,
-                    } => {
-                        let light = stage.new_value(SpotLight {
-                            position: Vec3::ZERO,
-                            direction: Vec3::NEG_Z,
-                            inner_cutoff: inner_cone_angle,
-                            outer_cutoff: outer_cone_angle,
-                            color,
-                            intensity,
-                        });
-                        (light.id().into(), LightDetails::Spot(light))
-                    }
-                };
                 let node_index = *light_index_to_node_index.get(&gltf_light.index()).context(
                     MissingCameraSnafu {
                         index: gltf_light.index(),
                     },
                 )?;
-                let node_transform = node_transforms
+
+                let mut node_transform = node_transforms
                     .get(&node_index)
                     .context(MissingNodeSnafu { index: node_index })?
                     .clone();
-                light.transform = node_transform.global_transform_id();
-
-                let light = stage.new_value(light);
-                lights.push(GltfLight {
-                    details,
-                    node_transform,
-                    light,
-                });
+                node_transform.move_gpu_to_slab(stage.lighting().slab());
+
+                let color = Vec3::from(gltf_light.color()).extend(1.0);
+                let intensity = gltf_light.intensity();
+                let light_bundle = match gltf_light.kind() {
+                    gltf::khr_lights_punctual::Kind::Directional => {
+                        stage.lighting().new_directional_light(
+                            DirectionalLightDescriptor {
+                                direction: Vec3::NEG_Z,
+                                color,
+                                // TODO: Set a unit for lighting.
+                                // We don't yet use a unit for our lighting, and we should.
+                                // https://www.realtimerendering.com/blog/physical-units-for-lights/
+                                //
+                                // NOTE:
+                                // glTF spec [1] says directional light is in lux, whereas spot and point are
+                                // in candelas. I haven't really set a unit, it's implicit in the shader, but it seems we
+                                // can roughly get candelas from lux by dividing by 683 [2].
+                                // 1. https://github.com/KhronosGroup/glTF/blob/main/extensions/2.0/Khronos/KHR_lights_punctual/README.md
+                                // 2. https://depts.washington.edu/mictech/optics/me557/Radiometry.pdf
+                                intensity: intensity / 683.0,
```

Comment:
This change has affected the `stag::gltf_support::test::normal_mapping_brick_sphere` test.

Reply:
Expectedly, multiplying the intensity by `683.0` fixes the test, so no biggy.